src/basic/socket-util.c

```diff
@@ -875,10 +872,7 @@ ssize_t receive_one_fd_iov(
                 int flags,
                 int *ret_fd) {
 
-        union {
-                struct cmsghdr cmsghdr;
-                uint8_t buf[CMSG_SPACE(sizeof(int))];
-        } control = {};
+        CMSG_BUFFER_TYPE(CMSG_SPACE(sizeof(int))) control;
```

Comment:
This one is used for recvmsg, and cmsg(3) does not say anything about initialization. But please confirm that it is dropped on purpose.

Reply:
my original intention was to leave initialization exactly as it was, but i fucked that up.

I figure we should add initialization to all sendmsg cases and drop it from all recvmsg cases. will do.